timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/LockWatchValueIntegrationTest.java

```diff
@@ -471,6 +487,61 @@ public void lateAbortingTransactionDoesNotFlushValuesToCentralCache() {
         });
     }
 
+    @Test
+    public void valueStressTest() {
+        createTransactionManager(1.0);
+        int numThreads = 200;
+        int numTransactions = 10_000;
+
+        ExecutorService executor = Executors.newFixedThreadPool(numThreads);
+        List<Exception> failedTransactions = new ArrayList<>();
+
+        for (int attempt = 0; attempt < numTransactions; ++attempt) {
+            executor.execute(() -> {
+                try {
+                    randomTransactionTask();
+                } catch (Exception e) {
+                    if (!(e instanceof TransactionFailedRetriableException)) {
+                        failedTransactions.add(e);
+                        throw e;
+                    }
+                }
+            });
+        }
+
+        try {
+            executor.shutdown();
+            executor.awaitTermination(60, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException(e);
+        }
+
+        assertThat(failedTransactions).isEmpty();
```

Comment:
Slightly different semantics, but I think it reads better:
```suggestion
        List<Future<?>> transactions = IntStream.range(0, numTransactions)
                .mapToObj(_num -> executor.submit(this::randomTransactionTask))
                .collect(Collectors.toList());

        for (Future<?> transaction : transactions) {
            try {
                transaction.get(60, TimeUnit.SECONDS);
            } catch (TransactionFailedRetriableException e) {
                // this is ok
            } catch (InterruptedException | TimeoutException e) {
                fail("Transaction took too long", e);
            } catch (ExecutionException e) {
                fail("Encountered nonretriable exception", e);
            }
        }

        executor.shutdown();
```